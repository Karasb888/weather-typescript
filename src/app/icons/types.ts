export interface IconsProps {
    width: string;
    height: string;
    viewBox: string;
    svgClass: string;
    title?: string;
};