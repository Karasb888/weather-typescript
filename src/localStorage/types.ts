export enum LocalStorageActionTypes {
    ADD_FAVORITE_CITY ='ADD_FAVORITE_CITY',
    REMOVE_FAVORITE_CITY = 'REMOVE_FAVORITE_CITY'
};

export enum LocalStorageNames {
    favoriteCities = 'favoriteCities'
};

export interface LocalStorageAction {
    type: LocalStorageActionTypes;
    payload: any;
};