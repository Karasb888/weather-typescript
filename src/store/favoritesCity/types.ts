export interface FavoriteCityObj {
    id: number;
    name: string;
};

export interface FavoriteCityState {
    citiesList: Array<FavoriteCityObj>;
};

export enum FavoritesCityActionTypes {
    SET_FAVORITE_CITY = 'SET_FAVORITE_CITY',
    REMOVE_FAVORITE_CITY = 'REMOVE_FAVORITE_CITY'
};