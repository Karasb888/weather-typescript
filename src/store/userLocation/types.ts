export interface GeoIpApiResponseType {
    status: string;
    country: string;
    countryCode: string;
    region: string;
    regionName: string;
    city: string;
    zip: number;
    lat: number;
    lon: number;
    timezone: string;
    isp: string;
    org: string;
    as: string;
    query: string;
};

export type FetchStatusType = 'success' | 'pending' | 'error' | 'none';

export interface LocationInfo {
    ip: string;
    city: string;
    country: string;
};

export interface UserLocationState {
    locationInfo: LocationInfo | null;
    fetchStatusUserLocation: FetchStatusType;
};

export enum UserLocationActionTypes {
    SET_USER_LOCATION_INFO = 'SET_USER_LOCATION_INFO',
    SET_FETCH_STATUS_USER_LOCATION = 'SET_FETCH_STATUS_USER_LOCATION'
};