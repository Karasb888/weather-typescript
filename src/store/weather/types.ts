// Api response types
export type TempObj<T> = {
    [P in keyof T]: T[P]
};

export type WeatherListItemObj = {
    id: number,
    main: string,
    description: string,
    icon: string
};

export type WeatherListObj = {
    dt: number,
    dt_txt: string,
    temp: TempObj<number>,
    pressure: number,
    humidity: number,
    weather: Array<WeatherListItemObj>,
    speed: number,
    deg: number,
    clouds: number,
    snow: number
};

export type CityObj = {
    geoname_id: number,
    name: string,
    lat: number,
    lon: number,
    country: number,
    iso2: string,
    type: string,
    population: number
};

export type WeatherResponseObj = {
    cod: number,
    message: string | number
    city: CityObj,
    cnt: number,
    list: Array<WeatherListObj>
};

//===================================================

export type NewWeatherDayObj = {
    date: string,
    dayTimestamp: number,
    timeDayWeather: Array<WeatherListObj>
};

export type WeatherCity = Array<NewWeatherDayObj> | null;
export type WeatherCityAndDate = Array<WeatherListObj> | null;
export type FetchStatusType = 'success' | 'pending' | 'error' | 'none';
export type City = CityObj | null;

export enum WeatherActionTypes {
    SET_CITY = 'SET_CITY',
    SET_CITY_WEATHER = 'SET_CITY_WEATHER',
    SET_CITY_AND_DATE_WEATHER = 'SET_CITY_AND_DATE_WEATHER',
    SET_FETCH_STATUS_WEATHER_CITY = 'SET_FETCH_STATUS_WEATHER_CITY'
};

export interface WeatherState {
    city: City,
    weatherCity: WeatherCity,
    fetchStatusWeatherCity: FetchStatusType,
    weatherCityAndDate: WeatherCityAndDate
};
