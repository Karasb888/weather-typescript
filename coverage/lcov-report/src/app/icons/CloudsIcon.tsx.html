
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/app/icons/CloudsIcon.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/app/icons</a> CloudsIcon.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react';
import { IconsProps } from './types';
&nbsp;
const CloudsIcon: React.FC&lt;IconsProps&gt; = ({ height, width, viewBox, svgClass }) =&gt; {
    return (
        &lt;svg
            height={height}
            viewBox={viewBox}
            width={width}
            className={svgClass}
            xmlns="http://www.w3.org/2000/svg"
        &gt;
            &lt;path d="m375.835938 112.957031c-5.851563 0-11.691407.582031-17.425782 1.742188-4.324218-21.582031-18.304687-39.992188-37.933594-49.957031-19.625-9.964844-42.738281-10.382813-62.714843-1.136719-18.078125-49.796875-73.101563-75.507813-122.898438-57.429688s-75.507812 73.105469-57.429687 122.898438c-43.621094 1.378906-78.078125 37.484375-77.4257815 81.121093.6562495 43.640626 36.1835935 78.691407 79.8281255 78.761719h296c48.597656 0 88-39.398437 88-88 0-48.601562-39.402344-88-88-88zm0 0" fill="#a3d4f7"/&gt;
        &lt;/svg&gt;
    );
};
&nbsp;
export default CloudsIcon;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sat Feb 22 2020 13:02:08 GMT+0200 (GMT+02:00)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    