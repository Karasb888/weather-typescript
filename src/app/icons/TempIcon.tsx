import * as React from 'react';
import { IconsProps } from './types';

const TempIcon: React.FC<IconsProps> = ({ height, width, viewBox, svgClass }) => {
    return (
        <svg
            id="Layer_3"
            enableBackground="new 0 0 64 64"
            height={height}
            viewBox={viewBox}
            width={width}
            className={svgClass}
            xmlns="http://www.w3.org/2000/svg"
        >
            <g>
            <circle cx="29" cy="35" fill="#c7e2fc" r="27"/>
                <path d="m3.214 43.029c-.789-2.536-1.214-5.233-1.214-8.029 0-2.062.231-4.07.669-6h6.331c2.209 0 4 1.791 4 4 0 2.209 1.791 4 4 4h1.42c2.038 0 3.75 1.533 3.976 3.558l.409 3.679c.104.94-.633 1.763-1.579 1.763-1.466 0-2.149 1.817-1.046 2.783l.771.675c2.343 2.05 5.818 2.127 8.248.182.53-.424 1.123-.761 1.759-.999l1.172-.439c3.284-1.232 6.957.336 8.339 3.561.35.816.531 1.695.531 2.584v4.846c-3.615 1.797-7.689 2.807-12 2.807-1.361 0-2.698-.101-4.005-.295v.001c-.099-2.989-1.707-5.733-4.283-7.278l-3.507-2.104c-1.44-.864-2.604-2.121-3.355-3.623-1.747-3.494-5.317-5.701-9.223-5.701h-1.422z" fill="#78b75b" />
                <path d="m46 14v10c0 1.105-.895 2-2 2h-2.5c-.828 0-1.5.672-1.5 1.5 0 .842-.693 1.521-1.539 1.499-.823-.021-1.461.736-1.461 1.559v1.942c0 .828.672 1.5 1.5 1.5 1.995 0 3.601 1.67 3.495 3.688-.099 1.893-1.798 3.312-3.693 3.312h-6.802c-1.933 0-3.5-1.563-3.5-3.825v-1.674c0-1.968-1.622-3.434-3.589-3.502-1.438-.05-2.567-1.314-2.394-2.799.15-1.282 1.327-2.2 2.617-2.2h.366 7c1.105 0 2-.895 2-2 0-1.091-.874-1.978-1.959-2-1.109-.022-2.041-.812-2.041-1.92v-3.102c-1.208-.913-2-2.347-2-3.977v-6.001l-.044.02c.347-.013.695-.02 1.044-.02 6.438 0 12.349 2.253 16.989 6.014z" fill="#78b75b"/>
                <path d="m13 24 3-3 5-1v-4l3-3-3-3v-4l-5-1-3-3-3 3-5 1v4l-3 3 3 3v4l5 1z" fill="#f7d881"/>
                <circle cx="13" cy="13" fill="#c16b3f" r="6"/>
                <path d="m7 13c0 1.088.294 2.105.8 2.985 4.002-.159 7.2-3.444 7.2-7.485 0-.406-.041-.801-.103-1.189-.597-.199-1.233-.311-1.897-.311-3.314 0-6 2.686-6 6z" fill="#e39854"/>
                <path d="m60 50.106v-23.106c0-2.761-2.239-5-5-5-2.761 0-5 2.239-5 5v23.106c-1.533 1.566-2.34 3.844-1.864 6.296.531 2.732 2.731 4.933 5.463 5.463 4.481.867 8.401-2.541 8.401-6.865 0-1.906-.764-3.632-2-4.894z" fill="#f7d881"/>
                <path d="m57 52.779v-24.779c0-1.105-.895-2-2-2-1.105 0-2 .895-2 2v24.779c-.73.657-1.145 1.656-.953 2.754.212 1.212 1.203 2.21 2.415 2.42 1.894.329 3.538-1.12 3.538-2.953 0-.885-.39-1.672-1-2.221z" fill="#e47c6e"/>
                <path d="m53 29h4v2h-4z" fill="#c66b60"/>
            </g>
        </svg>
    );
};

export default TempIcon;
