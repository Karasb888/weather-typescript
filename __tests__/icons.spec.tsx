import * as React from 'react';
import CloudsIcon from '../src/app/icons/CloudsIcon';
import ErrorIcon from '../src/app/icons/ErrorIcon';
import TempFeelsIcon from '../src/app/icons/TempFeelsIcon';
import TempIcon from '../src/app/icons/TempIcon';
import WindIcon from '../src/app/icons/WindIcon';
import { IconsProps } from '../src/app/icons/types';
import { shallow, ShallowWrapper } from 'enzyme';

const props: IconsProps = {
    width: '20',
    height: '20',
    svgClass: 'testClass',
    viewBox: '0 0 20 20'
};
let wrappers: Array<ShallowWrapper<IconsProps>>;

beforeEach(() => {
    wrappers = [
        shallow(<WindIcon {...props} />),
        shallow(<CloudsIcon {...props} />),
        shallow(<TempFeelsIcon {...props} />),
        shallow(<ErrorIcon {...props} />),
        shallow(<TempIcon {...props} />)
    ];
});

describe('Project svg icons', () => {
    it('renders correctly', () => {
        wrappers.forEach((wrapper) => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
