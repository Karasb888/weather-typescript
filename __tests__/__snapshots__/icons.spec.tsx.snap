// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project svg icons renders correctly 1`] = `
<svg
  className="testClass"
  height="20"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="m416 240h-384a16 16 0 0 1 0-32h384a16 16 0 0 0 0-32 16 16 0 0 1 0-32 48 48 0 0 1 0 96z"
    fill="#82cdaa"
  />
  <path
    d="m416 272h-384a16 16 0 0 1 0-32h384a48 48 0 0 0 0-96 16 16 0 0 0 0 32 16 16 0 0 1 0 32 48 48 0 0 1 0-96 80 80 0 0 1 0 160z"
    fill="#329682"
  />
  <path
    d="m192 176h-160a16 16 0 0 1 0-32h160a16 16 0 0 0 0-32 16 16 0 0 1 0-32 48 48 0 0 1 0 96z"
    fill="#82cdaa"
  />
  <path
    d="m192 208h-160a16 16 0 0 1 0-32h160a48 48 0 0 0 0-96 16 16 0 0 0 0 32 16 16 0 0 1 0 32 48 48 0 0 1 0-96 80 80 0 0 1 0 160z"
    fill="#329682"
  />
  <path
    d="m336 432a16 16 0 0 1 0-32 16 16 0 0 0 0-32h-144a16 16 0 0 1 0-32h144a48 48 0 0 1 0 96z"
    fill="#82cdaa"
  />
  <path
    d="m336 464a48 48 0 0 1 0-96 16 16 0 0 1 0 32 16 16 0 0 0 0 32 48 48 0 0 0 0-96h-144a16 16 0 0 1 0-32h144a80 80 0 0 1 0 160z"
    fill="#329682"
  />
  <g
    fill="#c39b69"
  >
    <path
      d="m40 336h8a48 48 0 0 1 48 48v8a8 8 0 0 1 -8 8h-8a48 48 0 0 1 -48-48v-8a8 8 0 0 1 8-8z"
    />
    <path
      d="m328 64h8a48 48 0 0 1 48 48v8a8 8 0 0 1 -8 8h-8a48 48 0 0 1 -48-48v-8a8 8 0 0 1 8-8z"
      transform="matrix(0 1 -1 0 448 -256)"
    />
    <path
      d="m424 288h8a48 48 0 0 1 48 48v8a8 8 0 0 1 -8 8h-8a48 48 0 0 1 -48-48v-8a8 8 0 0 1 8-8z"
    />
  </g>
</svg>
`;

exports[`Project svg icons renders correctly 2`] = `
<svg
  className="testClass"
  height="20"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="m375.835938 112.957031c-5.851563 0-11.691407.582031-17.425782 1.742188-4.324218-21.582031-18.304687-39.992188-37.933594-49.957031-19.625-9.964844-42.738281-10.382813-62.714843-1.136719-18.078125-49.796875-73.101563-75.507813-122.898438-57.429688s-75.507812 73.105469-57.429687 122.898438c-43.621094 1.378906-78.078125 37.484375-77.4257815 81.121093.6562495 43.640626 36.1835935 78.691407 79.8281255 78.761719h296c48.597656 0 88-39.398437 88-88 0-48.601562-39.402344-88-88-88zm0 0"
    fill="#a3d4f7"
  />
</svg>
`;

exports[`Project svg icons renders correctly 3`] = `
<svg
  className="testClass"
  enableBackground="new 0 0 64 64"
  height="20"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="#e6e9ed"
  >
    <path
      d="m10.505 44.432c0 1.101-.891 2-2 2-1.101 0-2-.899-2-2 0-1.11 2-3 2-3s2 1.889 2 3z"
    />
    <path
      d="m55.505 32.422v3h-6v-3c0-1.65 1.35-3 3-3 1.66 0 3 1.349 3 3z"
    />
    <path
      d="m38.505 44.432c0 1.101-.891 2-2 2-1.101 0-2-.899-2-2 0-1.11 2-3 2-3s2 1.889 2 3z"
    />
  </g>
  <path
    d="m28.495 43.962c-1.801.93-3.841 1.46-6 1.46-2.16 0-4.2-.53-6-1.46-4.16-2.17-7-6.521-7-11.54v-4l3-3s4.569 2 10 2c5.43 0 10-2 10-2l3 3v4c0 5.019-2.841 9.37-7 11.54z"
    fill="#eac6bb"
  />
  <path
    d="m32.495 25.422-.003.001v6.999c0 5.02-2.841 9.37-7 11.54-1.377.711-2.896 1.183-4.499 1.368.493.057.994.092 1.502.092 2.159 0 4.199-.53 6-1.46 4.159-2.17 7-6.521 7-11.54v-4z"
    fill="#d3b1a9"
  />
  <path
    d="m28.495 43.962v5.46l-3 2-3-1-3 1-3-2v-5.46c1.8.93 3.84 1.46 6 1.46 2.159 0 4.199-.53 6-1.46z"
    fill="#87bcf4"
  />
  <path
    d="m55.505 42.432v9.8c-.88-.52-1.91-.81-3-.81-1.03 0-2 .26-2.851.72l-.149.084v-9.794-7.01h6z"
    fill="#da4453"
  />
  <path
    d="m55.505 52.231c1.79 1.03 3 2.971 3 5.19 0 3.32-2.681 6-6 6-3.311 0-6-2.68-6-6 0-2.28 1.27-4.26 3.149-5.28.851-.46 1.82-.72 2.851-.72 1.09.001 2.12.291 3 .81z"
    fill="#ed5565"
  />
  <path
    d="m39.495 56.422v6c0 .56-.45 1-1 1h-32c-.551 0-1-.44-1-1v-6c0-3.87 3.13-7 7-7h4l3 2 3-1 3 1 3-2h4c3.869 0 7 3.13 7 7z"
    fill="#5d9cec"
  />
  <path
    d="m32.495 49.422h-4l-.116.077h1.113c3.869 0 7 3.13 7 7v6c0 .421-.254.773-.614.923h2.617c.55 0 1-.44 1-1v-6c0-3.87-3.131-7-7-7z"
    fill="#4a89dc"
  />
  <path
    d="m12.495 25.422-3 3v-5c0-6.63 5.369-12 12-12h8v4c3.31 0 6 2.689 6 6v7l-3-3s-4.57 2-10 2c-5.431 0-10-2-10-2z"
    fill="#965353"
  />
  <path
    d="m29.495 18.422c1.654 0 3 1.346 3 3v4l3 3v-7c0-3.311-2.69-6-6-6v-4h-3v4c0 1.657 1.343 3 3 3z"
    fill="#844747"
  />
  <path
    d="m22.523 56.422c-.553 0-1.005-.447-1.005-1s.442-1 .994-1h.011c.552 0 1 .447 1 1s-.448 1-1 1z"
    fill="#87bcf4"
  />
  <path
    d="m22.523 61.423c-.553 0-1.005-.447-1.005-1s.442-1 .994-1h.011c.552 0 1 .447 1 1s-.448 1-1 1z"
    fill="#87bcf4"
  />
  <path
    d="m15.496 8.422c-.321 0-.637-.154-.83-.441-.309-.458-.188-1.079.271-1.388.16-.126.506-.563-.244-1.575-.714-.961-.984-1.883-.804-2.739.218-1.035 1.021-1.518 1.112-1.569.482-.273 1.092-.104 1.363.375.269.472.111 1.069-.35 1.349-.025.019-.145.111-.172.275-.045.261.121.668.455 1.118.692.933.968 1.849.818 2.722-.187 1.085-.974 1.644-1.063 1.703-.169.114-.364.17-.556.17z"
    fill="#e6e9ed"
  />
  <path
    d="m22.497 8.422c-.321 0-.637-.154-.83-.441-.309-.458-.188-1.079.271-1.388.16-.126.506-.563-.244-1.575-.714-.961-.984-1.883-.804-2.739.218-1.035 1.021-1.518 1.112-1.569.483-.273 1.092-.104 1.363.375.269.472.111 1.069-.35 1.349-.025.019-.145.111-.172.275-.045.261.121.668.455 1.118.692.933.968 1.849.818 2.722-.187 1.085-.974 1.644-1.063 1.703-.169.114-.364.17-.556.17z"
    fill="#e6e9ed"
  />
  <path
    d="m29.498 8.422c-.321 0-.637-.154-.83-.441-.309-.458-.188-1.079.271-1.388.16-.126.506-.563-.244-1.575-.714-.961-.984-1.883-.804-2.739.218-1.035 1.021-1.518 1.112-1.569.482-.273 1.092-.104 1.363.375.269.472.111 1.069-.35 1.349-.025.019-.145.111-.172.275-.044.26.122.667.456 1.117.692.933.968 1.849.818 2.722-.187 1.085-.974 1.644-1.063 1.703-.17.115-.365.171-.557.171z"
    fill="#e6e9ed"
  />
  <path
    d="m50.507 41.424h-1c-.001 0-.001 0-.002 0v1.008.992h.002 1c.553 0 1-.447 1-1s-.447-1-1-1z"
    fill="#d3374e"
  />
  <path
    d="m54.507 41.424c-.553 0-1 .447-1 1s.447 1 1 1h.998v-.992-1.008z"
    fill="#d3374e"
  />
</svg>
`;

exports[`Project svg icons renders correctly 4`] = `
<svg
  className="testClass"
  enableBackground="new 0 0 512 512"
  height="20"
  version="1.1"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M322.939,62.642l178.737,309.583C508.231,383.578,512,396.74,512,410.791
                c0,42.67-34.592,77.264-77.264,77.264H256L194.189,256L256,23.946C284.62,23.946,309.587,39.519,322.939,62.642z"
    fill="#3B4145"
  />
  <path
    d="M189.061,62.642L10.323,372.225C3.769,383.578,0,396.74,0,410.791
                c0,42.67,34.592,77.264,77.264,77.264H256V23.946C227.38,23.946,202.413,39.519,189.061,62.642z"
    fill="#525A61"
  />
  <path
    d="M474.913,387.678L296.177,78.098c-8.056-13.959-22.849-22.767-38.848-23.22l152.869,402.275h24.539
                c25.559,0,46.358-20.798,46.358-46.358C481.095,402.677,478.952,394.683,474.913,387.678z"
    fill="#FFB751"
  />
  <path
    d="M444.853,387.678c3.492,7.005,5.336,14.999,5.336,23.117c0,25.559-17.935,46.358-39.992,46.358
                H77.264c-25.559,0-46.358-20.799-46.358-46.358c0-8.118,2.143-16.112,6.181-23.117l178.736-309.58
                c8.283-14.34,23.674-23.251,40.177-23.251c0.443,0,0.886,0.01,1.329,0.031c13.732,0.536,26.414,9.323,33.326,23.22L444.853,387.678z
                "
    fill="#FFD764"
  />
  <path
    d="M256,354.131v51.509c14.227,0,25.755-11.528,25.755-25.755
                C281.755,365.659,270.227,354.131,256,354.131z"
    fill="#3B4145"
  />
  <path
    d="M256,354.131c2.843,0,5.151,11.528,5.151,25.755c0,14.227-2.308,25.755-5.151,25.755
                c-14.227,0-25.755-11.528-25.755-25.755C230.245,365.659,241.773,354.131,256,354.131z"
    fill="#525A61"
  />
  <path
    d="M256,132.646V323.23c14.227,0,25.755-11.538,25.755-25.755V158.401
                C281.755,144.174,270.227,132.646,256,132.646z"
    fill="#3B4145"
  />
  <path
    d="M256,132.646c2.843,0,5.151,11.528,5.151,25.755v139.074c0,14.216-2.308,25.755-5.151,25.755
                c-14.227,0-25.755-11.538-25.755-25.755V158.401C230.245,144.174,241.773,132.646,256,132.646z"
    fill="#525A61"
  />
</svg>
`;

exports[`Project svg icons renders correctly 5`] = `
<svg
  className="testClass"
  enableBackground="new 0 0 64 64"
  height="20"
  id="Layer_3"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <circle
      cx="29"
      cy="35"
      fill="#c7e2fc"
      r="27"
    />
    <path
      d="m3.214 43.029c-.789-2.536-1.214-5.233-1.214-8.029 0-2.062.231-4.07.669-6h6.331c2.209 0 4 1.791 4 4 0 2.209 1.791 4 4 4h1.42c2.038 0 3.75 1.533 3.976 3.558l.409 3.679c.104.94-.633 1.763-1.579 1.763-1.466 0-2.149 1.817-1.046 2.783l.771.675c2.343 2.05 5.818 2.127 8.248.182.53-.424 1.123-.761 1.759-.999l1.172-.439c3.284-1.232 6.957.336 8.339 3.561.35.816.531 1.695.531 2.584v4.846c-3.615 1.797-7.689 2.807-12 2.807-1.361 0-2.698-.101-4.005-.295v.001c-.099-2.989-1.707-5.733-4.283-7.278l-3.507-2.104c-1.44-.864-2.604-2.121-3.355-3.623-1.747-3.494-5.317-5.701-9.223-5.701h-1.422z"
      fill="#78b75b"
    />
    <path
      d="m46 14v10c0 1.105-.895 2-2 2h-2.5c-.828 0-1.5.672-1.5 1.5 0 .842-.693 1.521-1.539 1.499-.823-.021-1.461.736-1.461 1.559v1.942c0 .828.672 1.5 1.5 1.5 1.995 0 3.601 1.67 3.495 3.688-.099 1.893-1.798 3.312-3.693 3.312h-6.802c-1.933 0-3.5-1.563-3.5-3.825v-1.674c0-1.968-1.622-3.434-3.589-3.502-1.438-.05-2.567-1.314-2.394-2.799.15-1.282 1.327-2.2 2.617-2.2h.366 7c1.105 0 2-.895 2-2 0-1.091-.874-1.978-1.959-2-1.109-.022-2.041-.812-2.041-1.92v-3.102c-1.208-.913-2-2.347-2-3.977v-6.001l-.044.02c.347-.013.695-.02 1.044-.02 6.438 0 12.349 2.253 16.989 6.014z"
      fill="#78b75b"
    />
    <path
      d="m13 24 3-3 5-1v-4l3-3-3-3v-4l-5-1-3-3-3 3-5 1v4l-3 3 3 3v4l5 1z"
      fill="#f7d881"
    />
    <circle
      cx="13"
      cy="13"
      fill="#c16b3f"
      r="6"
    />
    <path
      d="m7 13c0 1.088.294 2.105.8 2.985 4.002-.159 7.2-3.444 7.2-7.485 0-.406-.041-.801-.103-1.189-.597-.199-1.233-.311-1.897-.311-3.314 0-6 2.686-6 6z"
      fill="#e39854"
    />
    <path
      d="m60 50.106v-23.106c0-2.761-2.239-5-5-5-2.761 0-5 2.239-5 5v23.106c-1.533 1.566-2.34 3.844-1.864 6.296.531 2.732 2.731 4.933 5.463 5.463 4.481.867 8.401-2.541 8.401-6.865 0-1.906-.764-3.632-2-4.894z"
      fill="#f7d881"
    />
    <path
      d="m57 52.779v-24.779c0-1.105-.895-2-2-2-1.105 0-2 .895-2 2v24.779c-.73.657-1.145 1.656-.953 2.754.212 1.212 1.203 2.21 2.415 2.42 1.894.329 3.538-1.12 3.538-2.953 0-.885-.39-1.672-1-2.221z"
      fill="#e47c6e"
    />
    <path
      d="m53 29h4v2h-4z"
      fill="#c66b60"
    />
  </g>
</svg>
`;
